test/Feature/RpcSpec.hs

```diff
@@ -258,6 +268,12 @@ spec actualPgVersion =
       it "cannot return composite type in hidden schema" $
         post "/rpc/ret_point_3d" [json|{}|] `shouldRespondWith` 401
 
+      when (actualPgVersion >= pgVersion110) $
+        it "returns domain of composite type" $
+          post "/rpc/ret_composite_domain" [json|{}|] `shouldRespondWith`
+            [json|[{"x": 10, "y": 5}]|]
```

Comment:
Hmm.. but why should this return an array instead of an object? Shouldn't the output depend on the function `setof`?

If `setof` array, else object? This is the behavior I'm seeing on 7.0.1, with this change I see we always return an array.

Reply:
Absolutely, this should return an object. But this is fixed in part2 of #1615. In 7.0.1 only scalars are treated properly to not return an array. But composite types are not. The fix in here is just to treat domains of composites the same as composites.

So, 7.0.1 does the following:
- `RETURNS composite` returns array
- `RETURNS domain_of_composite` returns object

This fix:
- `RETURNS composite` returns array
- `RETURNS domain_of_composite` returns array

#1615:
- `RETURNS composite` returns object
- `RETURNS domain_of_composite` returns object

Reply:
Got it! Just checked 7.0.1 and that holds. Then this new test is good :+1:.